### PR TITLE
Add attest protobuf

### PR DESCRIPTION
**Pull request by nick-mobilecoin**

### CODE REVIEW

**Comment by nick-mobilecoin** [comment]
Current dependencies on/for this PR:
* main
  * **PR #2** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/probufs/2" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/mobilecoinfoundation/probufs/2?utm_source=stack-comment).

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -7,6 +7,13 @@ on:
   pull_request:
 
 jobs:
+  buf-lint:
```
Comment (nick-mobilecoin) [inline .github/workflows/ci.yaml:10]: buf-breaking doesn't work until we have proto files in the repo :(
So once this lands then I can add the buf-breaking ci check

File: attestapis/buf.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
+# AttestAPIs
+
+The APIs required to interact with the `AttestService`.
```
Comment (nick-mobilecoin) [inline attestapis/buf.md]: I'm up for suggestions on these docs.

Also you can see the rendering [here](https://github.com/mobilecoinfoundation/probufs/blob/nick/attest-proto/attestapis/buf.md)

File: attestapis/buf.yaml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+version: v1
+name: buf.build/nick-mc/attestapis
```
Comment (nick-mobilecoin) [inline attestapis/buf.yaml]: we'll need to decide where we publish to.

This PR shouldn't be merged until this is agreed on

**Review by jcape — CHANGES_REQUESTED** [review]
We should split types and services, I dislike "attestapis" as a name, and also I'd prefer to have the package end up being called `mobilecoinfoundation.attestation.v1` or something similar.

File: attestapis/buf.yaml (hunk 3 above)
Comment (nick-mobilecoin) [reply attestapis/buf.yaml]: updated name

File: foundation/mobilecoinfoundation/attestation/v1/attest.proto [hunk 4]
```diff
@@ -0,0 +1,132 @@
+syntax = "proto3";
+
+// The APIs required to interact with the `AttestService`.
```
Comment (nick-mobilecoin) [inline foundation/mobilecoinfoundation/attestation/v1/attest.proto]: per https://docs.buf.build/bsr/documentation#package-documentation, package documentation goes in the alphabetically first module prior to the package directive

File: foundation/buf.lock [hunk 5]
```diff
@@ -0,0 +1,7 @@
+# Generated by buf. DO NOT EDIT.
```
Comment (nick-mobilecoin) [inline foundation/buf.lock]: I believe the `foundation` was the repo/module name that was agreed to go into, https://buf.build/mobilecoin
unless we create a new `mobilecoinfoundation` owner

**Review by awygle — APPROVED** [review]
This matches what we discussed in the meeting.

File: foundation/buf.lock (hunk 5 above)
Comment (nick-mobilecoin) [reply foundation/buf.lock]: resolved we're creating a mobilecoinfoundation owner

File: mobilecoinfoundation/attestation/v1/attest.proto [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+//         Enclave-->>Client: AttestedMessge
+//     end
+// ```
+// 
+// ## Enclave to Enclave
+// 
+// ```mermaid
+// sequenceDiagram
+//     autonumber
+//     Enclave_1->>+Enclave_2: MutualAttestRequest
```
Comment (jcape) [inline mobilecoinfoundation/attestation/v1/attest.proto:31]:
```suggestion
//     Enclave_1->>+Enclave_2: MutualAttestRequest
//     Enclave_2->>Enclave_2: Verify MutualAttestRequest contents
```

**Review by jcape — APPROVED** [review]
LGTM, one comment

File: mobilecoinfoundation/attestation/v1/attest.proto (hunk 6 above)
Comment (nick-mobilecoin) [reply mobilecoinfoundation/attestation/v1/attest.proto:31]: Updated, also fixed spelling of `Messge` on line 37

**Pull request merged**